templatefuncs.go

```diff
@@ -159,6 +175,62 @@ func regexpReplaceAllTemplateFunc(expr, repl, s string) string {
 	return regexp.MustCompile(expr).ReplaceAllString(s, repl)
 }
 
+// reverseTemplateFunc is the core implementation of the `reverse`
+// template function.
+func reverseTemplateFunc(list []any) []any {
+	listcopy := append([]any(nil), list...)
+	slices.Reverse(listcopy)
+	return listcopy
+}
+
+// sortTemplateFunc is the core implementation of the `sort` template function.
+func sortTemplateFunc(list []any) any {
+	if len(list) < 2 {
+		return list
+	}
+
+	firstElemType := reflect.TypeOf(list[0])
+
+	for _, elem := range list[1:] {
+		if reflect.TypeOf(elem) != firstElemType {
+			return list
+		}
+	}
+
+	switch firstElemType.Kind() { //nolint:exhaustive
+	case reflect.Int:
+		return convertAndSortSlice[int](list)
+	case reflect.Int8:
+		return convertAndSortSlice[int8](list)
+	case reflect.Int16:
+		return convertAndSortSlice[int16](list)
+	case reflect.Int32:
+		return convertAndSortSlice[int32](list)
+	case reflect.Int64:
+		return convertAndSortSlice[int64](list)
+	case reflect.Uint:
+		return convertAndSortSlice[uint](list)
+	case reflect.Uint8:
+		return convertAndSortSlice[uint8](list)
```

Comment:
This will not work as expected, as `[]byte` is an alias for `[]uint8` and in other template functions we treat `[]byte`s as `string`s.